pkg/kube/controllers/extendedstatefulset/extendedstatefulset_controller.go

```diff
@@ -124,7 +124,26 @@ func AddExtendedStatefulSet(ctx context.Context, config *config.Config, mgr mana
 
 	// Watch Secrets referenced by the ExtendedStatefulSet
 	secretPredicates := predicate.Funcs{
-		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		CreateFunc: func(e event.CreateEvent) bool {
+
+			o := e.Object.(*corev1.Secret)
+			shouldProcessEvent := vss.IsVersionedSecret(*o)
+			reconciles, err := reference.GetReconciles(ctx, mgr.GetClient(), reference.ReconcileForExtendedStatefulSet, o, true)
+			if err != nil {
+				ctxlog.Errorf(ctx, "Failed to calculate reconciles for secret '%s': %v", o.Name, err)
+			}
+
+			if shouldProcessEvent && len(reconciles) > 0 {
+				ctxlog.NewPredicateEvent(e.Object).Debug(
```

Comment:
Other controllers do not log in the predicate if the watch has a `EnqueueRequestsFromMapFunc`, because that's also logging. Logging in line 170 should be enough.

Reply:
Done.